aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -23,6 +24,88 @@ void where_cpu(
         ret_val = cond_val ? self_val : other_val;
       });
 }
+
+template<template<typename T> class Comparator, typename scalar_out, typename scalar>
+struct CmpOp {
+  static void apply(at::Tensor& ret, const at::Tensor& self, const at::Tensor& other) {
+    at::CPU_tensor_apply3<scalar_out, scalar, scalar>(ret, self, other,
+        [](scalar_out& ret_val, const scalar& self_val, const scalar& other_val) {
+          ret_val = at::convert<scalar_out>(Comparator<scalar>()(self_val, other_val));
+        }
+    );
+  }
+
+  static void apply(at::Tensor& ret, const at::Tensor& self, at::Scalar other) {
```

Comment:
When is this overload used? Shouldn't we always use `CmpOpScalar` in this case?

Reply:
It's called in CmpOpScalar.